Sources/SourceControl/GitRepository.swift

```diff
@@ -33,9 +33,59 @@ public class GitRepositoryProvider: RepositoryProvider {
 
     /// Reference to process set, if installed.
     private let processSet: ProcessSet?
-
-    public init(processSet: ProcessSet? = nil) {
+    /// The path to the directory where all cached git repositories are stored.
+    private let cachePath: AbsolutePath
+    /// The maximum size of the cache in bytes.
+    private let maxCacheSize: UInt64
+
+    public init(processSet: ProcessSet? = nil,
+                cachePath: AbsolutePath = AbsolutePath(localFileSystem.homeDirectory, ".swiftpm/cache/repositories"),
+                maxCacheSize: UInt64 = 21_474_836_480) {
```

Comment:
It's not clear at a glance what the actual value is, could you express this as `20 * 1024 * 1024 * 1024`, maybe in a separate constant?